### DISCUSSION TITLE
search functionality

### DISCUSSION THREAD

**Discussion by osc2nuke** [sqlitebrowser/sqlitebrowser#3447] [Q&A] [answered]
I work on a Desktop app (laravel+nativePHP) what allows me to have some control over the machine (Windows 10)
I was very impressed how fast the program loads the table.
But more ... the search functionality.

Via a db request, the search query simply takes to long.

I wonder what is the code behind it.
Is that a command-line code what is executed?

Perhaps the question is a little strange, but i have good reasons to ask.

**Comment by chrisjlocke** [comment]
I'm not sure I understand the question.  You've developed a desktop app, but nativePHP makes it sound like a web application?
How does sqlite fit into this, and more importantly, how does DB Browser for SQLite fit into this?
Are you saying you've written a search query and its taking a long time in your application?
This issue section relates to issues with DB Browser for SQLite.  If you have a generic SQLite issue, the SQLite forum (www.sqlite.org) would be a better place to seek answers.

**Comment by osc2nuke** [comment]
I just wrote a Synopse on the NativePHP  Github discussion what give you some insight in what i'm looking for.

[For windows, Eloquent DB:: with sqlite3.exe?](https://github.com/NativePHP/laravel/discussions/190)

DB Browser has a search option build-in what was impressively fast, and i wondered if that was accomplished by default command-line, or some other way that would prob not be useful to me.

I also might got to enthusiastic to ask that here, i just became curious how fast the search worked in DB Browser.

**Comment by chrisjlocke** [comment, marked as answer]
SQLite is geared up for searching, and as everything is local, is pretty fast.
The biggest thing to bear in mind is that no two queries are the same.  It can make a difference if you're searching ten records or ten million.  Or if the columns have an index.  Or if you're searching for rows that begin with 'Hello' or ones that contain 'Hello'.  The former can use an index, while the latter has to search every row.  
The searching was done by throwing the SQLite library a query, and checking the response back.  So no 'command-line' per se, but talking to SQLite programmatically.

**Comment by osc2nuke** [comment]
I just noticed when i opened my database in DB Browser with 115.000 rows and scroll fast, it uses a kind of lazyLoading.

I think my prob is some how answered.

I apologize for interrupting into this discussion forum with my 'strange' question.
On the other hand ... if you don't ask ...... ( sometimes, be a little cheeky, most people are friendly)

**Reply by chrisjlocke** [reply]
> sometimes, be a little cheeky, most people are friendly

We try to be, and we encourage development.  Happy to help where we can.
We do tend to load in a small subset of records .. there is no way you could load 5,000,000 rows in a table in a speedy fashion.  Its silly really, as no one would want to load 5,000,000 rows in a table - what would be the point of scrolling through that many?  The early builds of DB Browser for SQLite did just that though .. oopsie.

**Comment by osc2nuke** [comment]
>> The early builds of DB Browser for SQLite did just that though .. oopsie.

I know my old friend ... i know, old wounds ...... We all have them, but we earned them in hard battle.

>>as no one would want to load 5,000,000 rows in a table 

The Gov wants that for +8.000.000.000
